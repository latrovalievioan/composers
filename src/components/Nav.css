.navigation {
    display: flex;
    flex-direction: row;
    padding: 0rem 1rem 0rem 1rem;
    border-radius: var(--border-radius);
    border-bottom: 2px solid var(--brown);
    box-shadow: 0px 30px 17px -20px var(--brown);
    margin-bottom: 2rem;
    align-content: center;
    justify-content: space-around;
    flex-wrap: wrap;
}

.nav-button {
    list-style: none;
    margin: 1rem;
}

.nav-button > a {
    color: var(--brown);
    text-decoration: none;
    font-size: 1.24rem;
}

.nav-button > a:hover {
    color: var(--orange)
}

.nav-button > .selected-link {
    color: var(--orange)
}