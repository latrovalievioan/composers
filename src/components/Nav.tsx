import { Link } from 'react-router-dom';
import React, { useState } from 'react';

import './Nav.css';

export const Nav = () => {
  const [currentPage, setCurrentPage] = useState('');

  console.log(currentPage);

  return (
    <nav className="navigation">
      {[
        { to: '/essentials', children: 'Essential Composers' },
        { to: '/popular', children: 'Popular Composers' },
        { to: '/favorites', children: 'Favorite Composers' },
      ].map((props) => (
        <li className="nav-button" key={props.to}>
          <Link
            {...props}
            onClick={() => setCurrentPage(props.to)}
            className={currentPage === props.to ? 'selected-link' : ''}
          />
        </li>
      ))}
    </nav>
  );
};
