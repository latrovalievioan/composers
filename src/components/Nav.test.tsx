import React from 'react';
import renderer from 'react-test-renderer';
import { Nav } from './Nav';
import { BrowserRouter } from 'react-router-dom';

it('renders correctly', () => {
  const nav = renderer
    .create(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>
    )
    .toJSON();
  expect(nav).toMatchSnapshot();
});
