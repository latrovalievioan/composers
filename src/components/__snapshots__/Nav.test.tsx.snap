// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navigation"
>
  <li
    className="nav-button"
  >
    <a
      className=""
      href="/essentials"
      onClick={[Function]}
    >
      Essential Composers
    </a>
  </li>
  <li
    className="nav-button"
  >
    <a
      className=""
      href="/popular"
      onClick={[Function]}
    >
      Popular Composers
    </a>
  </li>
  <li
    className="nav-button"
  >
    <a
      className=""
      href="/favorites"
      onClick={[Function]}
    >
      Favorite Composers
    </a>
  </li>
</nav>
`;
